contracts/AxelarGateway.sol

```diff
@@ -202,40 +199,51 @@ contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         uint256 adminCount = _getAdminCount(epoch);
         results = new address[](adminCount);
 
-        for (uint256 i; i < adminCount; ++i) {
+        for (uint256 i; i != adminCount; ) {
             results[i] = _getAdmin(epoch, i);
+
+            unchecked {
+                ++i;
+            }
         }
     }
 
     /*******************\
     |* Admin Functions *|
     \*******************/
 
-    function setTokenDailyMintLimits(string[] calldata symbols, uint256[] calldata limits) external override onlyAdmin {
-        if (symbols.length != limits.length) revert InvalidSetDailyMintLimitsParams();
+    function setTokenDailyMintLimits(string[] calldata symbols, uint256[] calldata limits) external payable override onlyAdmin {
+        uint256 symbolsLength = symbols.length;
+
+        if (symbolsLength != limits.length) revert InvalidSetDailyMintLimitsParams();
 
-        for (uint256 i = 0; i < symbols.length; i++) {
+        for (uint256 i; i != symbolsLength; ) {
             string memory symbol = symbols[i];
             uint256 limit = limits[i];
 
             if (tokenAddresses(symbol) == address(0)) revert TokenDoesNotExist(symbol);
 
             _setTokenDailyMintLimit(symbol, limit);
+
+            unchecked {
+                ++i;
+            }
         }
     }
 
     function upgrade(
         address newImplementation,
         bytes32 newImplementationCodeHash,
         bytes calldata setupParams
-    ) external override onlyAdmin {
+    ) external payable override onlyAdmin {
```

Comment:
Why payable

Reply:
it saves like 21 gas because some checks are not run when calling this but I don't think that's enough justification for the confusion it adds.

Reply:
agree. I don't think it justifies at all if all we are doing here is saving 21 gas.